src/formats/internal/memreader.c

```diff
@@ -83,22 +91,37 @@ int8_t memread_byte(memreader *reader) {
 
 int16_t memread_word(memreader *reader) {
     int16_t r;
+#ifdef BIG_ENDIAN_BUILD
+    r = __builtin_bswap16(*((uint16_t *)(reader->buf + reader->pos)));
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
+#endif
     reader->pos += sizeof(r);
     return r;
 }
 
 int32_t memread_dword(memreader *reader) {
     int32_t r;
+#ifdef BIG_ENDIAN_BUILD
+    r = __builtin_bswap32(*((int32_t *)(reader->buf + reader->pos)));
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
+#endif
     reader->pos += sizeof(r);
     return r;
 }
 
 float memread_float(memreader *reader) {
     float r;
+#ifdef BIG_ENDIAN_BUILD
+    uint32_t fl;
+    memcpy(&fl, reader->buf + reader->pos, sizeof(fl));
+    fl = __builtin_bswap32(fl);
+    reader->pos += sizeof(fl);
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
     reader->pos += sizeof(r);
+#endif
     return r;
 }
```

Comment:
I think memread_float (and similar: memwrite_float, sd_read_float, etc..)
should be implemented in terms of their respective integer read-write functions.

for memread_float, this would look like:
```c
   uint32_t u = memread_dword(reader);
   float f;
   memcpy(&f, &u, 4);
   return f;
```

this way, they don't need to perform any byte swapping and can be focused on just the uint-to-float bit cast.

Reply:
I mean this code is tested and working. dropping the bswap would make it not work.

Reply:
memread_dword performs a byteswap.

memread_float is currently only used to read some tournament mode pilot variables we ignore, unsure what you mean by tested and working